database/postgres/dal/news.py

```diff
@@ -27,3 +30,117 @@ def fetch_news(self, order_by=None, limit=None, filters=None):
+            .join(union_query, News.news_id == union_query.c.news_id, isouter=True
+        )
+        .outerjoin(TeamInNews, News.news_id == TeamInNews.news_id))
+
+        all_info_about_news_by_period_of_time_with_user_interactions_with_them = (
+            all_info_about_news_with_user_interactions_with_them
+            .filter(
+                News.save_at >= period_of_time
+            )
+        )
+
+        return list(set(all_info_about_news_by_period_of_time_with_user_interactions_with_them.all()))
+
+
+    def data_frame_to_work_with_user_and_news_data(self, user_and_news: list[tuple]) -> pd.DataFrame.mask and pd.DataFrame.mask and pd.DataFrame:
+        user_and_news_details_df = pd.DataFrame(
+            user_and_news,
+            columns=['news_id', 'blob_id', 'sport_id', 'save_at', 'team_id', 'interest_rate_score', 'interaction_score']
+
+        )
+        user_and_news_details_df.set_index('news_id', inplace=True)
+
+        user_interact_with_this_news_mask = user_and_news_details_df[user_and_news_details_df['interaction_score'] != 0]
+        user_interact_with_this_news_mask = user_interact_with_this_news_mask[['sport_id']].copy()
+
+        user_not_interact_with_this_news_mask = user_and_news_details_df[user_and_news_details_df['interaction_score'] == 0]
+        user_not_interact_with_this_news_mask = user_not_interact_with_this_news_mask[['sport_id']].copy()
+
+        return user_interact_with_this_news_mask, user_not_interact_with_this_news_mask, user_and_news_details_df
+
+
+    def cleaning_duplicate_news_where_is_more_than_one_club(self, news_coefficients: pd.DataFrame) -> pd.DataFrame:
+        news_coefficients_without_duplicates = news_coefficients.groupby('news_id').agg({
+            'blob_id': 'first',
+            'interest_rate_score': 'first',
+            'interaction_score': 'first',
+            'time_score': 'first',
+            'sport_score': 'first',
+            'team_score': 'sum',
+        }).reset_index()
+
+        news_coefficients_without_duplicates = news_coefficients_without_duplicates.set_index('news_id')
+
+        return news_coefficients_without_duplicates
+
+
+    def assign_adjusted_scores_for_masks(
+            self,
+            news_coefficients_without_duplicates: pd.DataFrame,
+            user_interact_with_this_news_mask: pd.DataFrame.mask,
+            user_not_interact_with_this_news_mask: pd.DataFrame.mask,
+    ) -> pd.DataFrame.mask and pd.DataFrame.mask:
+
+        user_interact_with_this_news_mask[['adjusted_score','blob_id']] = news_coefficients_without_duplicates[['adjusted_score','blob_id']]
+        user_not_interact_with_this_news_mask[['adjusted_score','blob_id']] = news_coefficients_without_duplicates[['adjusted_score','blob_id']]
+        self.delete_dataframe(news_coefficients_without_duplicates)
+
+        return user_interact_with_this_news_mask, user_not_interact_with_this_news_mask
+
+
+    def delete_dataframe(self, dataframe):
```

Comment:
First - a private method for sure
Second - isn't there a better option? Like using "with" or smth